aptos-move/mvhashmap/src/lib.rs

```diff
@@ -163,23 +200,27 @@ impl<K: Hash + Clone + Eq, V: TransactionWrite> MVHashMap<K, V> {
     /// if the access path has never been written before.
     pub fn delete(&self, key: &K, txn_idx: TxnIndex) {
         // TODO: investigate logical deletion.
-        let mut map = self.data.get_mut(key).expect("Path must exist");
-        map.remove(&txn_idx);
+        let mut v = self.data.get_mut(key).expect("Path must exist");
+        v.versioned_map.remove(&txn_idx);
```

Comment:
What if after the deletion there is no delta there?
Should we track this and update contains_delta?

Reply:
it's fine, we will just look for deltas there in resolver, if we don't find, it's nothing.